types/ramda/index.d.ts

```diff
@@ -697,10 +698,10 @@ export function gte(a: number): (b: number) => boolean;
 /**
  * Returns whether or not an object has an own property with the specified name.
  */
-export function has<T>(__: Placeholder, obj: T): (s: string) => boolean;
-export function has<T>(__: Placeholder): (obj: T, s: string) => boolean;
-export function has<T>(s: string, obj: T): boolean;
-export function has(s: string): <T>(obj: T) => boolean;
+export function has(__: Placeholder, obj: unknown): (s: string) => boolean;
+export function has<P extends string>(__: Placeholder): (obj: unknown, s: P) => obj is ObjectHavingSome<P>;
+export function has<P extends string>(s: P, obj: unknown): obj is ObjectHavingSome<P>;
+export function has<P extends string>(s: P): (obj: unknown) => obj is ObjectHavingSome<P>;
```

Comment:
I have removed the `<T>` type variable here since I always got an error in the type predicate when it remained in the code. I also do not know which purpose this type variable has in the first place. However this might introduce a breaking change when some has used something like `const checkFunc = R.has<object>("name")` so that checkFunc has the signature `object => boolean`.

Reply:
Okay, I just recognized that `unknown => boolean` is compatible to any function `T => boolean` so that this should not be a problem.